### PR TITLE
ci: init

### PR DESCRIPTION
**Pull request by IgorShadurin**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-export { FairDataProtocol, AccountData, PersonalStorage }
+export { FDPStorage, AccountData, PersonalStorage }
 
 // for require-like imports
 declare global {
   interface Window {
     // bound as 'fdp' via Webpack
     fdp: {
-      FairDataProtocol: typeof import('./fair-data-protocol').FairDataProtocol
+      FDPStorage: typeof import('./fdp-storage').FDPStorage
```
Comment (nugaon) [inline src/index.ts:13]: I would change it to `FdpStorage` to meet camelCase conventions

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  BEE_PEER_DEBUG_API_URL: 'http://127.0.0.1:11635'
+  BEE_TEST_CHEQUEBOOK: true
+
+jobs:
+  nodejs:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```
Comment (nugaon) [inline .github/workflows/tests.yaml:30]: well, there is no long time support for node v12 already, but we should support for 18.x

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          fetch-depth: 1
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Auth to Github Package Docker Registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login https://docker.pkg.github.com -u ${GITHUB_ACTOR} --password-stdin
```
Comment (nugaon) [inline .github/workflows/tests.yaml:45]: it seems right for builds made by `ethersphere`, but we will have our own build by `fairDataSociety/bee-factory`

File: package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@fairdatasociety/fair-data-protocol",
+  "name": "@fairdatasociety/fdp-storage",
   "version": "1.0.0",
```
Comment (nugaon) [inline package.json:3]: version should be 0.1.0 maybe

**Review by nugaon — CHANGES_REQUESTED** [review]
set the `REPO_GHA_PAT` secret as well in the repo's settings

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (IgorShadurin) [reply .github/workflows/tests.yaml:30]: ok, I removed node 12, but not added 18. Due to https://github.com/fairDataSociety/fdp-storage/issues/6

File: .github/workflows/release_github.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GoogleCloudPlatform/release-please-action@v2
+        id: release
+        with:
+          token: ${{ secrets.REPO_GHA_PAT }}
+          release-type: node
+          package-name: '@fairdatasociety/fdp-storage'
+          bump-minor-pre-major: true
```
[the remark this answers is not in the document]
Comment (IgorShadurin) [reply .github/workflows/release_github.yaml:20]: > why did you define it?

Are you talking about `package-name: '@fairdatasociety/fdp-storage'`? Will it work without it?

[the remark this answers is not in the document]
Comment (nugaon) [reply .github/workflows/release_github.yaml:20]: yes

File: README.md [hunk 6]
```diff
@@ -23,45 +23,45 @@
...
 ```js
-const FDP = require('@fairdatasociety/fair-data-protocol');
+const FDP = require('@fairdatasociety/fdp-storage');
 ```
 
 ### Use in a browser with browserify, webpack or any other bundler
 
 ```js
-const FDP = require('@fairdatasociety/fair-data-protocol');
+const FDP = require('@fairdatasociety/fdp-storage');
```
Comment (nugaon) [inline README.md:46]: this is the same line like above. you could merge it into one section

**Review by nugaon — APPROVED** [review]
check my ci related comment, but anyway LGTM : )

File: .github/workflows/publish_npmjs.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
+      - run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      - uses: apexskier/github-release-commenter@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.REPO_GHA_PAT }}
+          comment-template: |
+            Release {release_link} addresses this.
```
Comment (nugaon) [inline .github/workflows/publish_npmjs.yaml:25]: why do we need it?

**Pull request merged**